cpp/src/parquet/column_writer.cc

```diff
@@ -2328,7 +2328,13 @@ std::shared_ptr<ColumnWriter> ColumnWriter::Make(ColumnChunkMetaDataBuilder* met
   const ColumnDescriptor* descr = metadata->descr();
   const bool use_dictionary = properties->dictionary_enabled(descr->path()) &&
                               descr->physical_type() != Type::BOOLEAN;
-  Encoding::type encoding = properties->encoding(descr->path());
+  Encoding::type default_encoding =
+      (descr->physical_type() == Type::BOOLEAN &&
+       properties->data_page_version() == ParquetDataPageVersion::V2)
```

Comment:
I'm not sure if we need to check `properties->version() != ParquetVersion::PARQUET_1_0`. parquet-mr does not have a way to set format version and always write 1 in the footer.